govcd/api.go

```diff
@@ -59,9 +58,19 @@ func (cli *Client) NewRequest(params map[string]string, method string, reqUrl ur
 		// to the processing function.
 		payload := ""
 		if req.ContentLength > 0 {
-			buf := new(bytes.Buffer)
-			buf.ReadFrom(body)
-			payload = buf.String()
+			// We try to convert body to a *bytes.Buffer
+			var ibody interface{}
+			ibody = body
+			bbody, ok := ibody.(*bytes.Buffer)
```

Comment:
You could use a reflection here instead to check the type.

Reply:
I could. The advantage would be that I could log the exact type so that I will have data for future improvements. Do we need to?

Reply:
Addressed all the requests